bin/check-marathon-task.rb

```diff
@@ -50,20 +50,35 @@ class MarathonTaskCheck < Sensu::Plugin::Check::CLI
   option :server, short: '-s SERVER', long: '--server SERVER', required: true
   option :port, short: '-p PORT', long: '--port PORT', default: 8080
   option :task, short: '-t TASK', long: '--task TASK', required: true
-  option :instances, short: '-i INSTANCES', long: '--instances INSTANCES', required: true, proc: proc(&:to_i)
+  option :instances, short: '-i INSTANCES', long: '--instances INSTANCES',
+                     required: true, proc: proc(&:to_i)
+  option :protocol, short: '-P PROTOCOL', long: '--protocol PROTOCOL',
+                    required: false, default: 'http'
+  option :username, short: '-u USERNAME', long: '--username USERNAME',
+                    required: false
+  option :password, long: '--password PASSWORD', required: false
 
   def run
-    if config[:instances] == 0
+    if config[:instances].zero?
       unknown 'number of instances should be an integer'
     end
 
+    if !config[:username].nil? && config[:password].nil? ||
+       config[:username].nil? && !config[:password].nil?
+      unknown 'You must provide both username and password'
```

Comment:
I would not say so personally as you are adding new functionality and I am suggesting how that feature should work. That being said it was a suggestion and not a requirement.